packages/admin/src/api.ts

```diff
@@ -66,10 +73,73 @@ export class AdminClient extends BaseApiClient {
     });
   }
 
-  public async createProposal(proposal: CreateProposalRequest): Promise<ProposalResponseWithUrl> {
+  // added separate from CreateProposalRequest type as the `simulate` boolean is contained within defender-client
+  public async createProposal(
+    proposal: CreateProposalRequest & { simulate?: boolean; overrideSimulationOpts?: SimulationTransaction },
+  ): Promise<ProposalResponseWithUrl> {
     return this.apiCall(async (api) => {
+      let simulation: SimulationResponse | undefined = undefined;
+      let simulationData = '0x';
+      const isBatchProposal = (contract: PartialContract | PartialContract[]): contract is PartialContract[] =>
+        isArray(contract);
+
+      // handle simulation checks before creating proposal
+      if (proposal.simulate) {
+        // we do not support simulating batch proposals from the client.
+        if (isBatchProposal(proposal.contract)) {
+          throw new Error(
+            'Simulating a batch proposal is currently not supported from the API. Use the Defender UI to manually trigger a simulation.',
+          );
+        }
+        const overrideData = proposal.overrideSimulationOpts?.transactionData.data;
+        simulationData = overrideData ?? '0x';
+        // only check if we haven't overridden the simulation data property
+        if (!overrideData) {
+          // Check if ABI is provided so we can encode the function
+          if (!proposal.contract.abi) {
+            // no ABI found, request user to pass in `data` in overrideSimulationOpts
+            throw new Error(
+              'Simulation requested without providing ABI. Please provide the contract ABI or use the `overrideSimulationOpts` to provide the data property directly.',
+            );
+          }
+          const contractInterface = new Interface(proposal.contract.abi);
+
+          // this is defensive and should never happen since createProposal schema validation will fail without this property defined.
+          if (!proposal.functionInterface) {
```

Comment:
I might be missing something but should this condition include overrideSimulationOpts ?

Reply:
No, overrideSimulationOpts should only be used in the `simulateProposal` call. With the exception of a reference in the preliminary checks for `data` since it can skip those checks if data was provided.